src/__experimental__/components/radio-button/radio-button-classic.style.js

```diff
@@ -40,12 +40,16 @@ export default ({
     border-color: rgb(175, 175, 175);
   }
 
-  ${checked && `
-    circle { fill: rgba(0, 0, 0, 0.85); }
-  `}
+  ${HiddenCheckableInputStyle}:checked + ${StyledCheckableInputSvgWrapper} circle {
+    fill: rgba(0, 0, 0, 0.85);
+  }
 
   ${disabled && css`
-    circle { fill: #${checked ? '8099a4' : 'e6ebed'}; }
+    circle { fill: #e6ebed; }
```

Comment:
A very minor comment: For consistency with the rest of the file, maybe this should be formatted like this?
```css
circle {
  fill: #e6ebed;
}
```